compiler/qsc_frontend/src/resolve.rs

```diff
@@ -569,6 +582,68 @@ impl Resolver {
                 scope.tys.insert(Rc::clone(&name.name), id);
                 scope.terms.insert(Rc::clone(&name.name), id);
             }
+            ast::ItemKind::Export(export) => {
```

Comment:
This kind of lumps the bind and resolve steps together which isn't a great idea IMO. Ideally, we bind all names first in one pass, then we resolve all references in another pass, which eliminates ordering issues. 

In trying to find a good example to articulate why I think this is a problem, I came across this one:

```qsharp
namespace Foo {
    export { Bar.Baz };
    export { Baz };
}

namespace Bar {
    operation Baz(): Unit {
    }
}
```

The first `export` statement imports `Baz` into the namespace, and the second export statement references it. Should this compile?

And if yes, should the following compile too? Declaration order doesn't matter for any of the other items, and I don't think it should matter for exports either.

```qsharp
namespace Foo {
    export { Baz };
             ^^^ name error
    export { Bar.Baz };
}

namespace Bar {
    operation Baz(): Unit {
    }
}
```

How about this?

```qsharp
namespace Foo {
    operation Main() : Unit {
        Baz(); // can reference Baz no problem
    }

    export { Baz };
             ^^^ name error

    open Bar;
    
}

namespace Bar {
    operation Baz(): Unit {
    }
}
```